### PR TITLE
Complying with turning ComplexesCategories into a suggested package in QPA

### PR DESCRIPTION
**Pull request by kamalsaleh**
https://github.com/homalg-project/HigherHomologicalAlgebra/issues/160

### CODE REVIEW

**Review by mohamed-barakat — APPROVED** [review]

File: FunctorCategories/gap/Functors.gi [hunk 1]
```diff
@@ -527,65 +527,69 @@ InstallMethod( EquivalenceFromFullSubcategoryOfProjectivesObjectsIntoAdditiveClo
     
 end );
 
-##
-BindGlobal( "SET_ISOMORPHISMS_BETWEEN_SOURCE_AND_IMAGE_OF_YONEDA_EMBEDDING_OF_SOURCE",
+if IsPackageMarkedForLoading( "ToolsForHigherHomologicalAlgebra", ">= 2023.03-01" ) then
```
Comment (mohamed-barakat) [inline FunctorCategories/gap/Functors.gi:530]: Is this enough? Don't you need `ComplexesCategories` in the code below? If yes, then please bump its version in the suggested dependency.

**Comment by mohamed-barakat** [comment]
`CatReps` needs to be updated as well. Please

* bump the version of `FunctorCatogories`
* update the dependency on `FunctorCatogories` in `CatReps`
* bump the version of `CatReps`

File: FunctorCategories/gap/Functors.gi (hunk 1 above)
Comment (kamalsaleh) [reply FunctorCategories/gap/Functors.gi:530]: Should be enough, these isomorphisms only require `AdditiveFunctorByTwoFunctions` which is implemented in `ToolsForHigherHomologicalAlgebra`.

Comment (mohamed-barakat) [reply FunctorCategories/gap/Functors.gi:530]: Then please add `ToolsForHigherHomologicalAlgebra` as a suggested package and put the version there and delete it here.

File: FunctorCategories/gap/Functors.gi [hunk 2]
```diff
@@ -527,65 +527,69 @@ InstallMethod( EquivalenceFromFullSubcategoryOfProjectivesObjectsIntoAdditiveClo
     
 end );
 
-##
-BindGlobal( "SET_ISOMORPHISMS_BETWEEN_SOURCE_AND_IMAGE_OF_YONEDA_EMBEDDING_OF_SOURCE",
+if IsPackageMarkedForLoading( "ToolsForHigherHomologicalAlgebra", "" ) then
```
Comment (zickgraf) [inline FunctorCategories/gap/Functors.gi:530]: I think strictly speaking one should keep the version check here. Otherwise one can theoretically load a very old version of `ToolsForHigherHomologicalAlgebra` which does not have the required functions manually before loading FunctorCategories and will get warnings.

Comment (kamalsaleh) [reply FunctorCategories/gap/Functors.gi:530]: True

Comment (kamalsaleh) [reply FunctorCategories/gap/Functors.gi:530]: I will wait until the ci-tests pass or fail and push again.

File: FiniteCocompletions/tst/100_LoadPackage.tst [hunk 3]
```diff
@@ -15,6 +15,8 @@ gap> LoadPackage( "FunctorCategories", false );
 true
 gap> LoadPackage( "LazyCategories", false );
 true
+gap> LoadPackage( "Locales", false );
+true
```
Comment (zickgraf) [inline FiniteCocompletions/tst/100_LoadPackage.tst:19]: The aim of making ComplexesCategories a suggested dependency was to avoid having to load Locales everywhere. So instead of loading Locales here, it should be possible to simply revert the changes in https://github.com/homalg-project/CategoricalTowers/pull/266.

Comment (kamalsaleh) [reply FiniteCocompletions/tst/100_LoadPackage.tst:19]: I loaded `Locales` to avoid touching the tests. It is now @mohamed-barakat 's call.

Comment (mohamed-barakat) [reply FiniteCocompletions/tst/100_LoadPackage.tst:19]: Loading `Locales` in the tests should be ok.

Comment (zickgraf) [reply FiniteCocompletions/tst/100_LoadPackage.tst:19]: > Loading `Locales` in the tests should be ok.

Let's discuss this verbally.

**Comment by mohamed-barakat** [comment]
Thank you very much @kamalsaleh.

File: FiniteCocompletions/tst/100_LoadPackage.tst (hunk 3 above)
Comment (mohamed-barakat) [reply FiniteCocompletions/tst/100_LoadPackage.tst:19]: > Let's discuss this verbally.

As discussed, I will merge, then remove the `Locales` dependency in the tests of `Toposes` later.
